### PR TITLE
Document `error-pattern` header

### LINKED ISSUE #986: Document // error-pattern
Example: https://github.com/rust-lang/rust/pull/76143/files#diff-23706e664a2a576e6e031e0be43b04ff0f1916b93c2358b518d67e53aa92fbc3L1

It's used when you don't have a span available in the error but still want to test the output.

_Originally posted by @jyn514 in https://github.com/rust-lang/rust/pull/79621#discussion_r537715698_

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #986

### CODE REVIEW

File: src/tests/adding.md [hunk 1]
```diff
@@ -291,6 +294,37 @@ fn main() {
 //~| ERROR this pattern has 1 field, but the corresponding tuple struct has 3 fields [E0023]
 ```
 
+#### When error line cannot be specified
+
+Let's this about this test:
```
Comment (LeSeulArtichaut) [inline src/tests/adding.md:299]: Typo?

File: src/tests/adding.md [hunk 2]
```diff
@@ -291,6 +294,37 @@ fn main() {
...
+```rust,ignore
+fn main() {
+    let a: *const [_] = &[1, 2, 3];
+    unsafe {
+        let _b = (*a)[3];
+    }
+}
+```
+
+We want to ensure this shows "index out of bounds" but we cannot use the `ERROR` annotation
```
Comment (LeSeulArtichaut) [inline src/tests/adding.md:310]: Maybe specify "at runtime" to explain why this has the `run-fail`? I'm not sure this is really necessary though.
```suggestion
We want to ensure this shows "index out of bounds" at runtime, but we cannot use the `ERROR` annotation
```

**Review by LeSeulArtichaut — APPROVED** [review]
Awesome, I didn't know this functionnality at all! ❤️
I left a few nits

File: src/tests/adding.md (hunk 2 above)
Comment (JohnTitor) [reply src/tests/adding.md:310]: Ah, copied/pasted headers from the original test but `run-fail` may be confusing. Just removed it.

File: src/tests/adding.md (hunk 1 above)
Comment (JohnTitor) [reply src/tests/adding.md]: Yeah, it is!

**Comment by LeSeulArtichaut** [comment]
Thanks!

**Pull request merged**